chain/src/Pos/Chain/Txp/Tx.hs

```diff
@@ -120,8 +130,57 @@ checkTx it =
                 ("output #"%int%" has invalid coin")
                 i
           )
+        -- The following rules check to see if we have passed a "cutoffEpoch"
+        -- after which we reject transactions larger than a size specified
+        -- in the `configuration.yaml` via the `TxValidationRules` struct.
+        , ( if currentEpoch > cutoffEpoch
+                then (fromIntegral $ tvrTxAttrSize txValRules)
+                      > unknownAttributesLength (_txAttributes it)
+                else True
+          , sformat
+                ("size of Tx unknown attributes in input #"%int%" is too large")
+                i
+          )
+        , ( if currentEpoch > cutoffEpoch
+                then all ( < (fromIntegral $ tvrAddrAttrSize txValRules))
+                         (map unknownAttributesLength txOutAddrAttribs)
+                else True
+          , sformat
+                ("size of Address unknown attributes in input #"%int%" is too large")
+                i
+          )
         ]
-
+    currentEpoch = tvrCurrentEpoch txValRules
+    cutoffEpoch = tvrAddrAttrCutoff txValRules
+    txOutAddresses = map txOutAddress (NE.toList (_txOutputs it))
+    txOutAddrAttribs = map addrAttributes txOutAddresses
+
+-- | Because there is no limit on the size of Attributes
+-- (which allows unecessary bloating of the blockchain)
+-- this struct introduces limits configurable via the
+-- `configuration.yaml` file which are activated at
+-- the `tvrAddrAttrCutoff` epoch.
+data TxValidationRules = TxValidationRules
+    { tvrAddrAttrCutoff :: !EpochOrSlot
+    , tvrCurrentEpoch   :: !EpochOrSlot
+    , tvrAddrAttrSize   :: !Natural
+    , tvrTxAttrSize     :: !Natural
+    } deriving (Eq, Generic, Show)
+
+instance FromJSON TxValidationRules where
+    parseJSON = withObject "txValidationRules" $ \v -> TxValidationRules
+        <$> v .: "attribResrictEpoch"
+        <*> v .: "currEpoch"
+        <*> v .: "addrAttribSize"
+        <*> v .: "txAttribSize"
```

Comment:
👍 note: `FromJSON` instance for `TxValidationRules` fails if the key or fields aren't present [(`withObject` docs)](https://hackage.haskell.org/package/aeson-1.4.2.0/docs/Data-Aeson-Types.html#v:withObject). After some discussion about defaults and magic values (see earlier PR review) it was decided that a standard no-default fail-if-not-present parser would be better.